document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/UnnestExpression.java

```diff
@@ -0,0 +1,31 @@
+package org.hypertrace.core.documentstore.expression.impl;
+
+import com.google.common.base.Preconditions;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.hypertrace.core.documentstore.expression.type.FromTypeExpression;
+import org.hypertrace.core.documentstore.parser.FromTypeExpressionVisitor;
+
+/**
+ * This expression allows expanding an array field to a set of rows
+ *
+ * <p><code>
+ *  UnnestExpression.of(IdentifierExpression.of("array_col")) </code>
+ */
+@Value
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class UnnestExpression implements FromTypeExpression {
```

Comment:
Is unnest an accepted teminology? 

May be spread, or destruct? (Both borrowed from Javascript teminology)

Reply:
Different DBs use different terminologies. 
For example, 
`expode` in `Hive`
`unnest` in `Postgres`
`unwind` in `Mongo`

I prefer using a terminology from a DBMS rather than introducing a new terminology from Javascript/any other language.

Reply:
Most of the sql database use `unnest`, mongo uses the term `unwind`, some also use `explode`. 
However since our docstore language is follows sql database terminology, I went ahead with `unnest`.